bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java

```diff
@@ -149,20 +149,19 @@
      * @param secret openHAB's Secret to connect to the openHAB Cloud
      * @param remoteAccessEnabled Allow the openHAB Cloud to be used as a remote proxy
      * @param exposedItems Items that are made available to apps connected to the openHAB Cloud
-     *
      */
-    public CloudClient(String uuid, String secret, String baseURL, String localBaseUrl, boolean remoteAccessEnabled,
-            Set<String> exposedItems) {
+    public CloudClient(HttpClient httpClient, String uuid, String secret, String baseURL, String localBaseUrl,
+            boolean remoteAccessEnabled, Set<String> exposedItems) {
         this.uuid = uuid;
         this.secret = secret;
         this.baseURL = baseURL;
         this.localBaseUrl = localBaseUrl;
         this.remoteAccessEnabled = remoteAccessEnabled;
         this.exposedItems = exposedItems;
         runningRequests = new HashMap<Integer, Request>();
-        jettyClient = new HttpClient();
-        jettyClient.setMaxConnectionsPerDestination(HTTP_CLIENT_MAX_CONNECTIONS_PER_DEST);
-        jettyClient.setConnectTimeout(HTTP_CLIENT_TIMEOUT);
+        this.jettyClient = httpClient;
+        this.jettyClient.setMaxConnectionsPerDestination(HTTP_CLIENT_MAX_CONNECTIONS_PER_DEST);
+        this.jettyClient.setConnectTimeout(HTTP_CLIENT_TIMEOUT);
```

Comment:
Since this is a shared client, calling these setters will modify the behavior for all users of this client. The JavaDoc also [warns for not calling any setters](https://github.com/openhab/openhab-core/blob/2ace118b929653b0e0aeece5075b1255e9b26b43/bundles/org.openhab.core.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpClientFactory.java#L59-L65):

```java
    /**
     * Returns the shared Jetty http client. You must not call any setter methods or {@code stop()} on it.
     * The returned client is already started.
     *
     * @return the shared Jetty http client
     */
    HttpClient getCommonHttpClient();
```

So I think this JavaDoc suggests you should either use the default client "as is" or create your own client?

Reply:
We can handle the timeout for each `Request` separately. Thus there is no need to set it globally.

Now the question is, if we really need 200 connections at maximum. RFC 2616 suggests 2 but common browsers use 6.

Reply:
But wouldn't the values set by this openhabcloud add-on be overwritten if another add-on also starts calling these setters on the shared client with some other values?

Reply:
What I meant was to remove the calls to the setters and add the timeout to the `Request` object somewhere after line [329](https://github.com/openhab/openhab2-addons/blob/1256b94f6abc76bc62efbc221c3f9f3ae428fee0/bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java#L329) (e.g. request.timeout(HTTP_CLIENT_TIMEOUT);`

Regarding the possibility to call the setters for bindings see my https://github.com/openhab/openhab-core/issues/760#issuecomment-493697588.

Reply:
I submitted #5631.